### ISSUE TITLE
Lightning enabled pictureframe

### ISSUE THREAD

**Issue by StijnBTC** [satoshiradio/Projecten#1]

**Comment by lightningbrie** [comment]
Satoshi Radio youtube watcher requests a lightning invoice.
After the invoice is payed for, a picture appears on a screen in the Satoshi Radio studio.

The form where the invoice is requested should also offer to upload the picture.
We might want to enable the form only during certain times (e.g. during life shows).
Do we want to (later) allow video files to be uploaded and displayed?

**Comment by ronnyagous** [comment]
**More features:**

Niet vergeten: Moderator interface

Veiling voor 'zendtijd' bij veel animo? (bijvoorbeeld dutch auction)

Anderen kunnen de pictures 'liken' door een lightning betaling aan de maker. Bijvoorbeeld bij een goede authentieke meme. Incentive voor goede content.

**Comment by mlout** [comment]
> Niet vergeten: Moderator interface

Pictures first get put into a queue, only when they are approved by a moderator they appear on the frame.

> Veiling voor 'zendtijd' bij veel animo? (bijvoorbeeld dutch auction)

Maybe also automatically remove pictures from the slideshow once they have displayed X many times or have been showed for Y seconds/minutes,

Something like https://github.com/OpenFrameProject (although this project seems unmaintained) could be interesting, if we don't want to write everything our self.
Otherwise we could use something like fbi (https://www.raspberrypi-spy.co.uk/2017/02/how-to-display-images-on-raspbian-command-line-with-fbi/) or feh (https://bhoey.com/blog/photo-slideshows-using-raspberry-pi/) to display pictures if they are located on the pi.

**Comment by StijnBTC** [comment]
> Something like https://github.com/OpenFrameProject (although this project seems unmaintained) could be interesting, if we don't want to write everything our self.
Otherwise we could use something like fbi (https://www.raspberrypi-spy.co.uk/2017/02/how-to-display-images-on-raspbian-command-line-with-fbi/) or feh (https://bhoey.com/blog/photo-slideshows-using-raspberry-pi/) to display pictures if they are located on the pi.

We could also just make a browser run in kiosk mode. I think that's the easiest to set up

**Comment by niels-daniel** [comment]
> Otherwise we could use something like fbi (https://www.raspberrypi-spy.co.uk/2017/02/how-to-display-images-on-raspbian-command-line-with-fbi/) or feh (https://bhoey.com/blog/photo-slideshows-using-raspberry-pi/) to display pictures if they are located on the pi.

This sounds quite straightforward. 
If we create a folder where images are put in after moderation, showing them with FBI is just a single command. 

So steps could be:
1. Upload widget with Lighting payment (the current donation form should be a good start)
2. Images are stored in a “to be moderated” folder and in db.
3. Confirm/moderate widget (behind login obvious)
4. On approval move image from “to be moderated” to “live / display” folder (and adjust status in db)
5. When needed, turn on screen and run fbi command

Because we keep track of the images in the db, we can delete then after some time e.a.

**Comment by mlout** [comment]
> We could also just make a browser run in kiosk mode. I think that's the easiest to set up

Yes, good recommendation. I think this will also make things simpler, as we can have the whole program running in one place (satoshi radio webserver?).
I think the steps posted by @niels-daniel look good (only we host all the images also where the widgets are also running and just have a URL to display the slideshow).
